pkg/apis/config/v1alpha1/types.go

```diff
@@ -0,0 +1,148 @@
+// Copyright The NRI Plugins Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/instrumentation"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/log"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/resmgr/control"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/resmgr/policy/balloons"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/resmgr/policy/template"
+	"github.com/containers/nri-plugins/pkg/apis/config/v1alpha1/resmgr/policy/topologyaware"
+)
+
+// TopologyAwarePolicy represents the configuration for the topology-aware policy.
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +genclient
+type TopologyAwarePolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   TopologyAwarePolicySpec `json:"spec"`
+	Status ConfigStatus            `json:"status,omitempty"`
+}
+
+// TopologyAwarePolicySpec describes a topology-aware policy.
+type TopologyAwarePolicySpec struct {
+	Policy topologyaware.Config `json:"policy"`
```

Comment:
Now that `Common` was eliminated gave another thought to the `Policy` field and wonder if we should eliminate it, too

```suggestion
	topologyaware.Config `json:",inline"`
```

It would move all the policy config fields to the top (spec) level.. As an example, it would make the config yaml look like this

```yaml
apiVersion: config.nri/v1alpha1
kind: TopologyAwarePolicy
metadata:
  name: default
  namespace: kube-system
spec:
  pinCPU: true
  pinMemory: true
  reservedPoolNamespaces:
    - special-namespace
    - extra-reserved-namespace
  reservedResources:
    cpu: 750m
  log:
    klog:
      skip_headers: true
  instrumentation:
    reportPeriod: 60s
```

I'm a bit indeterminate on this one. as in my some earlier comment I said that the `policy` field makes sense. But now that I look at it, it doesn't necessarily so much. The object type name is already TopologyAware**Policy** so the `policy` field feels unnecessary and tautological. WDYT?

Reply:
> I'm a bit indeterminate on this one. as in my some earlier comment I said that the `policy` field makes sense. But now that I look at it, it doesn't necessarily so much. The object type name is already TopologyAware**Policy** so the `policy` field feels unnecessary and tautological. WDYT?

+1

Reply:
Changed all the former `Policy` fields inlined, dropping out `policy` from the configuration data on user-visible interfaces.